starter-code/src/viking.js

```diff
@@ -1,11 +1,111 @@
 // Soldier
-function Soldier() {}
+function Soldier(healthArg, strengthArg) {
+  this.health = healthArg;
+  this.strength = strengthArg;
+}
+
+Soldier.prototype.attack = function () {
+  return this.strength;
+};
+
+Soldier.prototype.receiveDamage = function (damage) {
+  this.health -= damage;
+};
 
 // Viking
-function Viking() {}
+function Viking(name, health, strength) {
+  Soldier.call(this, health, strength);
+  this.name = name;
+}
+
+Viking.prototype = Object.create(Soldier.prototype);
+
+Viking.prototype.receiveDamage = function (damage) {
+  this.health -= damage;
+  return (this.health > 0) ? this.name + " has received " + damage + " points of damage" : this.name + " has died in act of combat";
+}
+
+Viking.prototype.battleCry = function () {
+  return ("Odin Owns You All!");
+}
+
+
 
 // Saxon
-function Saxon() {}
+function Saxon(health, strength) {
+  Soldier.call(this, health, strength);
+}
+
+Saxon.prototype = Object.create(Soldier.prototype);
+
+Saxon.prototype.receiveDamage = function (damage) {
+  this.health -= damage;
+  return (this.health > 0) ? "A Saxon has received " + damage + " points of damage" : "A Saxon has died in combat";
+}
+
+
 
 // War
-function War() {}
+function War() {
+  this.vikingArmy = [];
+  this.saxonArmy = [];
+}
+
+War.prototype.addViking = function (Viking) {
+  this.vikingArmy.push(Viking);
+};
+
+War.prototype.addSaxon = function (Saxon) {
+  this.saxonArmy.push(Saxon);
+};
+
+War.prototype.vikingAttack = function () {
+  
+  function entierAleatoire(min, max) {
+    return Math.floor(Math.random() * (max - min + 1)) + min;
+  }
+
+  var RandomViking = this.vikingArmy[entierAleatoire(0, this.vikingArmy.length - 1)];
+  var RandomSaxon = this.saxonArmy[entierAleatoire(0, this.saxonArmy.length - 1)];
+
```

Comment:
instead of creating a function that you have to give 2 arguments and recall it to bring back a random number, you can just generate a random number based on the length of the array, save it in a variable and use the number every time as an index on the array you need to use.